subscription.go

```diff
@@ -31,3 +33,16 @@ func (sub *Subscription) Next(ctx context.Context) (*Message, error) {
 func (sub *Subscription) Cancel() {
 	sub.cancelCh <- sub
 }
+
+func (sub *Subscription) NextSubscribedPeer(ctx context.Context) (peer.ID, error) {
```

Comment:
maybe call this `NextPeer`.

Reply:
Sure and what would you like the next leave event to be called? I thought `NextPeerJoin` and `NextPeerLeave` might have a symmetry, but your call.